### PR TITLE
k8s: add ListConfigMaps API

### PR DESCRIPTION
**Pull request by scarlettperry**
### Description
PR adds a `ListConfigMaps` API
* unit tests added
* configmaps added to `kind.sh` for local testing

In a separate PR, will add APIs to get a ConfigMap by name and to delete a ConfigMap.
### Testing Performed
```
k get cm -n envoy-staging
NAME                       DATA   AGE
configmap-staging-test-1   1      8h
configmap-staging-test-2   1      8h
```

1. Get all ConfigMaps in a namespace
```
curl -X POST 'http://localhost:8080/v1/k8s/listConfigMaps' \
-H 'Content-Type: application/json' \
-d '{
    "clientset": "kind-clutch-local",
    "cluster": "kind-clutch-local",
    "namespace": "envoy-staging",
    "options": {}
}'
```

Response:
```
{"configMaps":[{"cluster":"kind-clutch-local","namespace":"envoy-staging","name":"configmap-staging-test-1","data":{"environment":"staging"},"annotations":{},"labels":{"app":"envoy"}},{"cluster":"kind-clutch-local","namespace":"envoy-staging","name":"configmap-staging-test-2","data":{"environment":"staging"},"annotations":{},"labels":{}}]}
```

2. Get ConfigMaps in a namespace, filter by label
```
curl -X POST 'http://localhost:8080/v1/k8s/listConfigMaps' \
-H 'Content-Type: application/json' \
-d '{
    "clientset": "kind-clutch-local",
    "cluster": "kind-clutch-local",
    "namespace": "envoy-staging",
    "options": {
        "labels": {
            "app": "envoy"
        }
    }
}'
```

Response:
```
{"configMaps":[{"cluster":"kind-clutch-local","namespace":"envoy-staging","name":"configmap-staging-test-1","data":{"environment":"staging"},"annotations":{},"labels":{"app":"envoy"}}]}
```

3. No ConfigMaps found
```
curl -X POST 'http://localhost:8080/v1/k8s/listConfigMaps' \
-H 'Content-Type: application/json' \
-d '{
    "clientset": "kind-clutch-local",
    "cluster": "kind-clutch-local",
    "namespace": "envoy-staging",
    "options": {
        "labels": {
            "app": "pagerduty"
        }
    }
}'
```

Response:
```
{"configMaps":[]}
```

### CODE REVIEW

File: api/k8s/v1/k8s.proto [hunk 1]
```diff
@@ -509,6 +517,35 @@ message DeleteCronJobRequest {
...
+  map<string, string> annotations = 5;
+  map<string, string> labels = 6;
+}
+
+message ListConfigMapsRequest {
+  string clientset = 1 [ (validate.rules).string = {min_bytes : 1} ];
+  string cluster = 2 [ (validate.rules).string = {min_bytes : 1} ];
+  string namespace = 3 [ (validate.rules).string = {min_bytes : 1} ];
+
+  ListOptions options = 4 [ (validate.rules).message = {required : true} ];
```
Comment (scarlettperry) [inline api/k8s/v1/k8s.proto:540]: for reference:
https://github.com/lyft/clutch/blob/main/api/k8s/v1/k8s.proto#L189

to be able to use this helper method https://github.com/lyft/clutch/blob/4d0b7faab3f2b5cc4c5b6bb9de84eb52b174d393/backend/service/k8s/meta.go#L12

File: backend/service/k8s/k8s.go [hunk 2]
```diff
@@ -80,6 +80,9 @@ type Service interface {
 	// CronJob management functions.
 	DescribeCronJob(ctx context.Context, clientset, cluster, namespace, name string) (*k8sapiv1.CronJob, error)
 	DeleteCronJob(ctx context.Context, clientset, cluster, namespace, name string) error
+
+	// ConfigMap management functions.
+	ListConfigMaps(ctx context.Context, clientset, cluster, namespace string, listOptions *k8sapiv1.ListOptions) ([]*k8sapiv1.ConfigMap, error)
```
Comment (mcutalo88) [inline backend/service/k8s/k8s.go:85]: To maintain our current patterns we should rename this to `DescribeConfigMap`, if you check out the other describes you will see they are all `List()` invocations. This allows us to search all namespaces, you can read more about this history here https://github.com/lyft/clutch/pull/271.

File: api/k8s/v1/k8s.proto [hunk 3]
```diff
@@ -509,6 +517,35 @@ message DeleteCronJobRequest {
...
+  option (clutch.api.v1.id).patterns = {
+    type_url : "clutch.k8s.v1.ConfigMap",
+    pattern : "{cluster}/{namespace}/{name}"
+  };
+
+  string cluster = 1;
+  string namespace = 2;
+  string name = 3;
+
+  map<string, string> data = 4;
```
Comment (rithujohn191) [inline api/k8s/v1/k8s.proto:530]: What data is intended to be put in this field?

File: backend/service/k8s/configmap.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	k8sapiv1 "github.com/lyft/clutch/backend/api/k8s/v1"
+)
+
+func (s *svc) ListConfigMaps(ctx context.Context, clientset, cluster, namespace string, listOptions *k8sapiv1.ListOptions) ([]*k8sapiv1.ConfigMap, error) {
+	cs, err := s.manager.GetK8sClientset(clientset, cluster, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := ApplyListOptions(listOptions)
```
Comment (rithujohn191) [inline backend/service/k8s/configmap.go:17]: This is not directly related to this PR, but currently `ApplyListOptions` does not support fieldselector. If we expand this method now we will be able to use this to get specific configmaps based on name, which is currently not possible. We should probably do a follow up PR to enable `ApplyListOptions` to support fieldselector

File: api/k8s/v1/k8s.proto (hunk 3 above)
Comment (scarlettperry) [reply api/k8s/v1/k8s.proto]: removed 👍

**Review by rithujohn191 — APPROVED** [review]
Looks great  👍  Just the one comment about the data field and applyListOptions :)

File: backend/service/k8s/configmap.go (hunk 4 above)
Comment (scarlettperry) [reply backend/service/k8s/configmap.go:17]: Agree, will create a ticket to follow-up on!

**Pull request merged**